pandas/tests/scalar/timestamp/test_constructors.py

```diff
@@ -599,21 +599,13 @@ def test_bounds_with_different_units(self):
     @pytest.mark.parametrize("arg", ["001-01-01", "0001-01-01"])
     def test_out_of_bounds_string_consistency(self, arg):
         # GH 15829
-        msg = "|".join(
-            [
-                "Cannot cast 1-01-01 00:00:00 to unit='ns' without overflow",
-                "Out of bounds nanosecond timestamp: 1-01-01 00:00:00",
-            ]
-        )
+        msg = "Cannot cast 0001-01-01 00:00:00 to unit='ns' without overflow"
         with pytest.raises(OutOfBoundsDatetime, match=msg):
             Timestamp(arg).as_unit("ns")
 
-        if arg == "0001-01-01":
-            # only the 4-digit year goes through ISO path which gets second reso
-            #  instead of ns reso
-            ts = Timestamp(arg)
-            assert ts.unit == "s"
-            assert ts.year == ts.month == ts.day == 1
+        ts = Timestamp(arg)
+        assert ts.unit == "s"
+        assert ts.year == ts.month == ts.day == 1
```

Comment:
nice! and also a reminder that if we have an `if` in a test, we should probably be explicit about the `else`, else it gets forgotten and stays in forever

I'm having a look and when this was fixed

Reply:
This was fixed in #51039

https://www.kaggle.com/code/marcogorelli/pandas-regression-example?scriptVersionId=124051955